src/stactools/sentinel2/mgrs.py

```diff
@@ -0,0 +1,210 @@
+"""Implements the :stac-ext:`MGRS Extension <mgrs>`."""
+
+from typing import Any, Dict, Optional, Union, cast
+from typing import Final, FrozenSet, Pattern
+import re
+
+import pystac
+from pystac.extensions.base import (
+    ExtensionManagementMixin,
+    PropertiesExtension,
+)
+# from pystac.extensions.base import (
+#     SummariesExtension,
+# )
+from pystac.extensions.hooks import ExtensionHooks
+
+SCHEMA_URI: Final[
+    str] = "https://stac-extensions.github.io/mgrs/v1.0.0/schema.json"
+PREFIX: Final[str] = "mgrs:"
+
+# Field names
+LATITUDE_BAND_PROP: Final[str] = PREFIX + "latitude_band"  # required
+GRID_SQUARE_PROP: Final[str] = PREFIX + "grid_square"  # required
+UTM_ZONE_PROP: Final[str] = PREFIX + "utm_zone"
+
+LATITUDE_BANDS: Final[FrozenSet[str]] = frozenset({
+    'C', 'D', 'E', 'F', 'G', 'H', 'J', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S',
+    'T', 'U', 'V', 'W', 'X'
+})
+
+UTM_ZONES: Final[FrozenSet[int]] = frozenset({
+    1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21,
+    22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40,
+    41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59,
+    60
+})
+
+GRID_SQUARE_REGEX: Final[str] = \
+    r"[ABCDEFGHJKLMNPQRSTUVWXYZ][ABCDEFGHJKLMNPQRSTUV](\d{2}|\d{4}|\d{6}|\d{8}|\d{10})?"
```

Comment:
s2 ids are only of the form "1AAA", but finer MGRS grids can be up to "1AAA1234567890"

Reply:
The mgrs extension doesn't support the finer grids either, it probably needs at least another field at some point, but isn't needed now for this.

Reply:
the STAC MGRS Extension isn't clear that it doesn't support it -- the grid_square field is just a string in both the text and json schema. If it's supposed to only be a 2 letter grid, we should restrict it as such. 

I think we should allow any of the finer grids. Either way, the MGRS json schema could be written a bit tighter.